swan-lake/learn-the-platform/ballerina-tooling/strand-dump-tool.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Strand dump tool
+description: Check out how you can dump and inspect the currently available strands of a Ballerina program.
+keywords: ballerina runtime, troubleshoot, strand dump, thread dump
+permalink: /learn/strand-dump-tool/
+active: strand-dump-tool
+redirect_from:
+  - /learn/strand-dump-tool
+---
+
+Ballerina runtime can have unexpected behaviors due to user code errors, bugs or issues with the running environment. 
+These will result in memory leaks, CPU spinning, runtime hangs, performance degradation or crashing with various errors. 
+Ballerina provides a tool to dump the status of currently running strands.
+
+## Strand dump
+
+Ballerina strand dump provides information on the available strands and strand groups during the execution of 
+a Ballerina program. This can be used to:
+
+- troubleshoot runtime errors,
+
+- find data races, race conditions, livelocks and deadlocks,
+
+- inspect strand and strand group status.
+
+>**Note:** Currently, this ability is only available in the operating systems where the `SIGTRAP` POSIX signal is 
+supported (`SIGTRAP` is not available on Windows).
+
+### Get the strand dump
+
+To get the strand dump when a Ballerina program is running, you need to know the process ID (PID) of the Ballerina 
+program. For that you can use the `jps` tool. Then you need to send the `SIGTRAP` signal to the process. The strand 
+dump will be produced to the standard output stream in the text format.
+
+As an example, consider the following Ballerina program.
+```ballerina
+import ballerina/lang.runtime;
+import ballerina/io;
+
+public function main() {
+    future<int> addResult = start addnum(1, 2);
+    int|error addition = wait addResult;
+    io:println(addition);
+}
+
+function addnum(int num1, int num2) returns int {
+
+    worker sender {
+        runtime:sleep(1000);
+        num1 -> receiver;
+    }
+
+    worker receiver returns int {
+        int firstNum = <- sender;
+        return num2 + firstNum;
+    }
+
+    int intResult = wait receiver;
+    return intResult;
+}
+```
+
+Run this Ballerina package using
+```
+$ bal run
```

Comment:
```suggestion
$ bal run
```
Sperate out the command and the CLI output of it to 2 code snippets.

Reply:
Did not do this because this is not done in other places also. E.g., https://ballerina.io/learn/get-started-with-ballerina/#run-the-package. If I do this, it is shown with a lot of space between the snippets.